test/e2e/windows/gmsa_full.go

```diff
@@ -0,0 +1,396 @@
+//    https://github.com/kubernetes-sigs/windows-gmsa/blob/master/scripts/GenerateCredentialSpecResource.ps1
+//    with the credential specs of that GMSA account, or type GMSACredentialSpec and named gmsa-e2e;
+//    and that manifest file has been written to C:\gmsa\gmsa-cred-spec-gmsa-e2e.yml
+//    on that same worker node.
+//  * the API has both MutatingAdmissionWebhook and ValidatingAdmissionWebhook
+//    admission controllers enabled.
+//  * the cluster comprises at least one Linux node that accepts workloads - it
+//    can be the master, but any other Linux node is fine too. This is needed for
+//    the webhook's pod.
+// All these assumptions are fulfilled by an AKS extension when setting up the AKS
+// cluster we run daily e2e tests against, but they do make running this test
+// outside of that very specific context pretty hard.
+
+package windows
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"github.com/pkg/errors"
+)
+
+const (
+	// gmsaFullNodeLabel is the label we expect to find on at least one node
+	// that is then expected to fulfill all the expectations explained above.
+	gmsaFullNodeLabel = "agentpool=windowsgmsa"
+
+	// gmsaCrdManifestPath is where we expect to find the manifest file for
+	// the GMSA cred spec on that node - see explanations above.
+	gmsaCrdManifestPath = `C:\gmsa\gmsa-cred-spec-gmsa-e2e.yml`
+
+	// gmsaCustomResourceName is the expected name of the GMSA custom resource
+	// defined at gmsaCrdManifestPath
+	gmsaCustomResourceName = "gmsa-e2e"
+
+	// gmsaWebhookDeployScriptURL is the URL of the deploy script for the GMSA webook
+	// TODO(wk8): we should pin versions.
+	gmsaWebhookDeployScriptURL = "https://raw.githubusercontent.com/kubernetes-sigs/windows-gmsa/master/admission-webhook/deploy/deploy-gmsa-webhook.sh"
+)
+
+var _ = SIGDescribe("[Feature:Windows] [Feature:WindowsGMSA] GMSA Full [Slow]", func() {
+	f := framework.NewDefaultFramework("gmsa-full-test-windows")
+
+	ginkgo.Describe("GMSA support", func() {
+		ginkgo.It("works end to end", func() {
+			defer ginkgo.GinkgoRecover()
+
+			ginkgo.By("finding the worker node that fulfills this test's assumptions")
+			nodes := findPreconfiguredGmsaNodes(f.ClientSet)
+			if len(nodes) != 1 {
+				framework.Skipf("Expected to find exactly one node with the %q label, found %d", gmsaFullNodeLabel, len(nodes))
+			}
+			node := nodes[0]
+
+			ginkgo.By("retrieving the contents of the GMSACredentialSpec custom resource manifest from the node")
+			crdManifestContents := retrieveCRDManifestFileContents(f, node)
```

Comment:
just a nit - if this isn't referenced, will the linter fail?

I think the test sufficiently validates that the CRD exists as-is, so maybe this needs to be `_ = ...` . If the CRD contents are wrong, then `createPodWithGmsa` will fail later

Reply:
This is referenced in `customResourceCleanup, err := createGmsaCustomResource(crdManifestContents)` below :)

Reply:
oh yeah, missed that, needed to hit ctrl-f again